src/main/java/com/github/checkstyle/regression/module/ModuleUtils.java

```diff
@@ -0,0 +1,172 @@
+            new HashMap<>();
+
+    /** Prevents instantiation. */
+    private ModuleUtils() {
+    }
+
+    /**
+     * Sets the map of full qualified name to module extract info with the given map.
+     * @param map the given map
+     */
+    public static void setNameToModuleExtratInfo(Map<String, ModuleExtractInfo> map) {
+        NAME_TO_MODULE_EXTRAT_INFO.clear();
+        NAME_TO_MODULE_EXTRAT_INFO.putAll(map);
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a checkstyle module.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle module
+     */
+    public static boolean isCheckstyleModule(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            returnValue = NAME_TO_MODULE_EXTRAT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a checkstyle utility class.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle utility class.
+     */
+    public static boolean isCheckstyleUtility(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            returnValue = !NAME_TO_MODULE_EXTRAT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a Java main source file.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a Java main source file.
+     */
+    private static boolean isJavaMainSource(GitChange change) {
+        final boolean returnValue;
+        final Matcher matcher = JAVA_SOURCE_PARTTEN.matcher(change.getPath());
+        if (matcher.find()) {
+            returnValue = "main".equals(matcher.group(1));
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a test of checkstyle module.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a test of checkstyle module
+     */
+    public static boolean isCheckstyleModuleTest(GitChange change) {
+        final boolean returnValue;
+        if (JAVA_SOURCE_PARTTEN.matcher(change.getPath()).find()) {
```

Comment:
This is `true` for test and main. It should only be accepting test. Make a `isJavaTestSource` and use that.